pkg/api/index.go

```diff
@@ -100,3 +107,32 @@ func addToIndex(ctx context.Context, key, value string) error {
 func storeAttestation(ctx context.Context, uuid string, attestation []byte) error {
 	return storageClient.StoreAttestation(ctx, uuid, attestation)
 }
+
+type Container interface {
+	Add([]string)
+	Result() []string
+}
+
+type Uniq map[string]struct{}
+
+func (u Uniq) Add(elements []string) {
```

Comment:
Nice implementation!

Question: If this is for `And` then won't we want an intersection? This seems to add any one of them, with a unique result.

For example, I expect the last query to be the intersection of the first two results, and only have 1 UUID
```
asraa@asraa1:~/git/rekor$ ./rekor search --public-key publickey.pem  --pki-format x509 --rekor_server http://localhost:3000
Found matching entries (listed by UUID):
64b1e670a1cf84b4d636d2d578fea277126279529759f537879796d8480f284f05708ba02b10ca38
64b1e670a1cf84b4af3fda3822b3db033f55be3af22324fa16381dc432d60d1ef1bd7ea7036538b1
asraa@asraa1:~/git/rekor$ ./rekor search --artifact COPYRIGHT.txt --rekor_server http://localhost:3000
Found matching entries (listed by UUID):
64b1e670a1cf84b47d1d6bedeb657dde5830a09aea72c528358786ee37183d7fb547f05692bb8af0
64b1e670a1cf84b4d636d2d578fea277126279529759f537879796d8480f284f05708ba02b10ca38
asraa@asraa1:~/git/rekor$ ./rekor search --public-key publickey.pem --operator and --artifact COPYRIGHT.txt  --pki-format x509 --rekor_server http://localhost:3000
Found matching entries (listed by UUID):
64b1e670a1cf84b47d1d6bedeb657dde5830a09aea72c528358786ee37183d7fb547f05692bb8af0
64b1e670a1cf84b4d636d2d578fea277126279529759f537879796d8480f284f05708ba02b10ca38
64b1e670a1cf84b4af3fda3822b3db033f55be3af22324fa16381dc432d60d1ef1bd7ea7036538b1
```

Reply:
I think intersection/union are not the correct words to use here. It's not a set. Basically, what we are returning is either the union of both values (i.e no duplicates), or just return all values, not minding the duplicates. I don't see a usecase for intersection.

Reply:
What is the use-case of not minding duplicates? For anyone iterating through results, that would be inefficient.

Yes! I thought intersection is the most useful: e.g. "I want signatures on package X signed by user Y."

Reply:
Yeah. That makes sense. I'll go about it this way, if there's no operator, it'll work like it does now, and return all values. Then we can use the operator for intersection or union.